### ISSUE TITLE
XFA forms still fail to render properly (or at all), even with (experimental) XFA support enabled.

### ISSUE THREAD

**Issue by sictransit** [mozilla/pdf.js#13508]
Attach (recommended) or Link to PDF file here:

[Market Authorisation Application](http://esubmission.ema.europa.eu/eaf/eAF_124.01/maa_human_v1.24.0.1.pdf) form from the European Medicines Agency.

Configuration:
- Web browser and its version:

```
Google Chrome | 91.0.4472.77&nbsp;(Officiell version)&nbsp;(64 bitar)&nbsp;(cohort: Stable)
Version | 1cecd5c8a856bc2a5adda436e7b84d8d21b339b6-refs/branch-heads/4472@{#1246}
```

- Operating system and its version: **Windows 10 (1909)**
- PDF.js version: Online [demo version](https://mozilla.github.io/pdf.js/web/viewer.html) + tested locally with v2.9.359

Steps to reproduce the problem:
1. enableXfa = true
2. load the PDF

What is the expected behavior? (add screenshot)

The expected behavior is that pdf.js renders, or at least tries to render the XFA form. I know it is still experimental, at best.

![image](https://user-images.githubusercontent.com/4610247/121042019-93f03b00-c7b3-11eb-964c-9464152ff936.png)

What went wrong? (add screenshot)

You do not get the standard "please wait" notice:

![image](https://user-images.githubusercontent.com/4610247/121042370-e6c9f280-c7b3-11eb-83f6-49c8ff11f9de.png)

Instead I'm seeing a different one:

![image](https://user-images.githubusercontent.com/4610247/121041965-83d85b80-c7b3-11eb-8118-30a30240b1a1.png)

Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension):
N/A

**Comment by sictransit** [comment]
Mea culpa! 

It seems I failed to enable XFA. 

The latest and greatest version of PDF.js actually renders the form. Not well, but that's another issue.

**Comment by marco-c** [comment]
@sictransit let's keep this open to track rendering this form as it is supposed to be rendered (it is a pretty complex one!)

**Comment by sictransit** [comment]
Fair enough! I'm preparing a filled out form now and I'll post an example here. It will take some time because, as you said, it's pretty complex.

The empty form actually renders more or lessOK, but I'm getting lots of exceptions with a real one.

**Comment by sictransit** [comment]
I've edited the issue description now, to better reflect the actual problem. I've also attached screenshots and a filled out form for test.
